ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -2010,10 +2000,7 @@ invalidBlockRejector tracer version getIsInvalidBlock getCandidate =
         mapM_ (uncurry disconnect)
           $ firstJust
                 (\hdr -> (hdr,) <$> isInvalidBlock (headerHash hdr))
-          $ (   case isPipeliningEnabled version of
-                    ReceivingTentativeBlocks    -> drop 1
-                    NotReceivingTentativeBlocks -> id
-            )
+          $ drop 1
```

Comment:
I think we'd want a comment here that calls out that this `drop 1` exists because of diffusion pipelining

Reply:
I guess we want more than simply a comment that states:

```
We drop the last ? header in the fragment because of diffusion pipelining
```

It'd be great if we could explain why we drop 1. I don't know if you could provide a couple of sentences @amesgen

Reply:
I think this is no longer relevant as @crocodile-dentist implemented the approach suggested in https://github.com/IntersectMBO/ouroboros-consensus/pull/1262#discussion_r1797178683.

I might add a more extensive comment why we drop exactly 1 header in a separate PR; this doesn't have to happen here.